dictionary.lisp

```diff
@@ -0,0 +1,120 @@
+#|
+ This file is a part of Qtools-UI
+ (c) 2019 Shirakumo http://tymoon.eu (shinmera@tymoon.eu)
+ Author: Michał "phoe" Herda <phoe@disroot.org>
+|#
+
+(in-package #:org.shirakumo.qtools.ui)
+(in-readtable :qtools)
+
+(define-widget dictionary (qwidget)
+  ((empty-browser-text :accessor empty-browser-text
+                       :initarg :empty-browser-text)
+   (not-found-text :accessor not-found-text
+                   :initarg :not-found-text))
+  (:default-initargs
+    :empty-browser-text
+    "<i><p align=center>Type your query below and hit Search.</p></i>"
+    :not-found-text
+    "<i><p align=center>The entry for \"~A\" was not found.</p><i>"))
+
+(define-subwidget (dictionary layout) (q+:make-qgridlayout)
+  (setf (q+:layout dictionary) layout
+        (q+:contents-margins layout) (values 0 0 0 0)))
+
+(define-subwidget (dictionary browser)
+    (make-instance 'browser :dictionary dictionary)
+  (q+:add-widget layout browser 0 0 1 2)
```

Comment:
I much prefer defining the layout last and adding the widgets in the layout's initialiser, as this is more layout-specific information.

Reply:
Fixed.